cpp/src_prims/random/rng.h

```diff
@@ -263,16 +263,19 @@ class Rng {
 
   /**
    * @brief Generate bernoulli distributed boolean array
-   * @tparam Type data type in which to compute the probabilities
+   *
+   * @tparam Type    data type in which to compute the probabilities
+   * @tparam OutType output data type
    * @tparam LenType data type used to represent length of the arrays
-   * @param ptr the output array
-   * @param len the number of elements in the output
-   * @param prob coin-toss probability for heads
-   * @param stream stream where to launch the kernel
+   *
+   * @param[out] ptr    the output array
+   * @param[in]  len    the number of elements in the output
+   * @param[in]  prob   coin-toss probability for heads
+   * @param[in]  stream stream where to launch the kernel
    */
-  template <typename Type, typename LenType = int>
-  void bernoulli(bool *ptr, LenType len, Type prob, cudaStream_t stream) {
-    randImpl<bool, Type>(
+  template <typename Type, typename OutType = bool, typename LenType = int>
+  void bernoulli(OutType *ptr, LenType len, Type prob, cudaStream_t stream) {
+    randImpl<OutType, Type>(
```

Comment:
Maybe this came in from a stray PR? But it looks clearly fine so no major issue to merge.

Reply:
Sorry, I forgot to describe this in our conversations earlier. It is deliberate. The existing `bernoulli` method used to always work on `bool` output array. I updated it to work on arbitrary types, as then I could just reuse this method from the accuracy test.